src/commands/flow/internal/spec/executeFlow.spec.ts

```diff
@@ -0,0 +1,200 @@
+import SASjs, { PollOptions } from '@sasjs/adapter/node'
+import { AuthConfig, Logger, LogLevel, ServerType, Target } from '@sasjs/utils'
+import { executeFlow } from '../executeFlow'
+import * as internalModule from '../'
+import csvColumns from '../csvColumns'
+
+const pollOptions: PollOptions = {
+  maxPollCount: 24 * 60 * 60,
+  pollInterval: 1000,
+  streamLog: false,
+  logFolderPath: 'fake/log/folder/path'
+}
+const fakeLogFileName = 'fakeLog.log'
+const fakeLogPath = `${pollOptions.logFolderPath}/${fakeLogFileName}`
+describe('executeFlow', () => {
+  const target: Target = {
+    name: 'test-target',
+    serverUrl: 'http://server.com',
+    serverType: ServerType.SasViya,
+    appLoc: '/test',
+    contextName: 'fakeContext'
+  } as any as Target
+  const sasjs = new SASjs()
+  jest
+    .spyOn(sasjs, 'fetchLogFileContent')
+    .mockImplementation(() => Promise.resolve(''))
+  process.logger = new Logger(LogLevel.Off)
+
+  beforeEach(() => {
+    jest
+      .spyOn(internalModule, 'saveLog')
+      .mockImplementation(() => Promise.resolve(fakeLogFileName))
+    jest
+      .spyOn(internalModule, 'saveToCsv')
+      .mockImplementation(() => Promise.resolve())
+    jest
+      .spyOn(internalModule, 'normalizeFilePath')
+      .mockImplementation((filePath: string) => filePath)
+  })
+
+  it('should execute flow with 2 successful jobs and 1 failing job', async () => {
+    const flowName = 'firstFlow'
+    const flow = {
+      name: flowName,
+      jobs: [
+        { location: 'jobs/testJob/job' },
+        { location: 'jobs/testJob/failingJob' },
+        { location: 'jobs/testJob/job' }
+      ],
+      predecessors: []
+    }
+
+    jest
+      .spyOn(sasjs, 'startComputeJob')
+      .mockImplementation(async (jobLocation: string) =>
+        jobLocation === '/test/jobs/testJob/job'
+          ? Promise.resolve({ job: { state: 'success' } })
+          : Promise.reject()
+      )
+
+    await executeFlow(
+      flow,
+      sasjs,
+      pollOptions,
+      target,
+      {} as any as AuthConfig,
+      ''
+    )
+
+    const csvDataSuccess = [
+      flowName,
+      'none',
+      `${target.appLoc}/${flow.jobs[0].location}`,
+      'success',
+      fakeLogPath,
+      ''
+    ]
+    const csvDataFailed = [
+      flowName,
+      'none',
+      `${target.appLoc}/${flow.jobs[1].location}`,
+      'failure',
+      fakeLogPath,
+      ''
+    ]
+
+    validatesaveToCsv([csvDataFailed, csvDataSuccess, csvDataSuccess])
+  })
+
+  it('should execute flow with 1 successful job and 1 job that does not exist', async () => {
+    const flowName = 'firstFlow'
+    const flow = {
+      name: flowName,
+      jobs: [
+        { location: 'jobs/testJob/job' },
+        { location: 'jobs/testJob/DOES_NOT_EXIST' }
```

Comment:
`jobs/testJob/DOES_NOT_EXIST` should be a variable that is also used in assertion

Reply:
fixed